lib/modules/manager/pep621/processors/uv.spec.ts

```diff
@@ -97,32 +95,100 @@ describe('modules/manager/pep621/processors/uv', () => {
         },
         {
           depName: 'dep2',
+          depType: 'tool.uv.sources',
+          datasource: 'git-refs',
+          packageName: 'https://github.com/foo/bar',
+          currentValue: undefined,
           skipReason: 'git-dependency',
         },
         {
           depName: 'dep3',
+          depType: 'tool.uv.sources',
           skipReason: 'path-dependency',
         },
         {
           depName: 'dep4',
+          depType: 'tool.uv.sources',
           skipReason: 'unsupported-url',
         },
         {
           depName: 'dep5',
+          depType: 'tool.uv.sources',
           skipReason: 'inherited-dependency',
         },
         {
           depName: 'dep6',
-          skipReason: 'invalid-dependency-specification',
```

Comment:
These `skipReason` shouldn't disappear

Reply:
The [new union](https://github.com/renovatebot/renovate/pull/31928/files#diff-060b5f839d75d04f183bebed7b71c536d693cfb10b610c4f4914a0b6fd7c8e17R59-R64) do not allow the invalid sources specifications [that were previously tested](https://github.com/renovatebot/renovate/pull/31928/files#diff-4e070e322f5eb5bb28ff9cb6c872a7f1c0dff9021ec92f4ae19971d20507f5a9L74-L75) anymore.
(This skip reason also was removed from [the processor](https://github.com/renovatebot/renovate/pull/31928/files#diff-d8da9aafff81cdba1a3495ed391d9536f840192d4d417ee7fea96447ca421c2fL58).)

Reply:
So UV basically maps falls back here to PyPi resolution?

Reply:
If the source spec is invalid, `uv` will fail to parse it and will not install the project

- `cowsay = { workspace = false }`
```
error: Failed to generate package metadata for `renovate-repro-uv-git-rev==0.1.0 @ virtual+.`
  Caused by: Failed to parse entry for: `cowsay`
  Caused by: `workspace = false` is not yet supported
``` 

- `cowsay = {  }`
```
error: Failed to parse: `pyproject.toml`
  Caused by: TOML parse error at line 10, column 10
   |
10 | cowsay = {  }
   |          ^^^^
expected one of `git`, `url`, `path`, `index`, or `workspace`
```

Reply:
I have meant is there isn't any definition in the sources.

Reply:
Ah yes, same as `dep1` indeed